### PR TITLE
Update pytorch-2.0 compatibility

### PR DESCRIPTION
**Pull request by Animatory**

### CODE REVIEW

File: torchok/constructor/constructor.py [hunk 1]
```diff
@@ -329,10 +329,8 @@ def _prepare_transforms_recursively(transforms: ListConfig) -> List[Union[A.Comp
             transform_name = transform_info.name
             transform_params = transform_info.get('params', dict())
 
-            if transform_name in ['Compose', 'OneOf', 'SomeOf', 'PerChannel', 'Sequential']:
+            if 'transforms' in transform_params:
```
Comment (VladVin) [inline torchok/constructor/constructor.py:332]: It's hard to understand the logic behind this. Why did you change the original implementation?

File: torchok/models/heads/detection/__init__.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import importlib
 
 has_mmcv = importlib.util.find_spec("mmcv")
-if has_mmcv is not None:
-    import torchok.models.heads.detection.fcos
-    import torchok.models.heads.detection.detr
+# ToDo: fix mmdet heads
```
Comment (VladVin) [inline torchok/models/heads/detection/__init__.py:4]: So, detection doesn't work?

File: torchok/tasks/base.py [hunk 3]
```diff
@@ -121,17 +122,18 @@ def on_predict_start(self) -> None:
         if self._hparams.task.load_checkpoint is not None:
             load_checkpoint(self, **self._hparams.task.load_checkpoint)
 
-    def training_step(self, batch: Dict[str, Union[torch.Tensor, int]], batch_idx: int) -> Dict[str, torch.Tensor]:
+    def training_step(self, batch: Dict[str, Union[Tensor, int]], batch_idx: int) -> Dict[str, Tensor]:
         """Complete training loop."""
         output = self.forward_with_gt(batch)
         total_loss, tagged_loss_values = self.losses(**output)
+        self.log("loss", total_loss, prog_bar=True, on_step=True)
```
Comment (VladVin) [inline torchok/tasks/base.py:129]: We don't usually log on each step. Why did you decide it to do for user?

**Review by VladVin — CHANGES_REQUESTED** [review]

File: torchok/constructor/config_structure.py [hunk 4]
```diff
@@ -134,17 +134,12 @@ class TaskParams:
 # Trainer parameters
 @dataclass
 class TrainerParams:
-    enable_checkpointing: bool = True
-    default_root_dir: Optional[str] = None
-    gradient_clip_val: Optional[float] = None
-    gradient_clip_algorithm: Optional[str] = None
+    accelerator: str = "auto"
+    strategy: str = "auto"
+    devices: str = "auto"
```
Comment (VladVin) [inline torchok/constructor/config_structure.py:139]: Shouldn't be the example configs updated after all these changes?

**Review by VladVin — CHANGES_REQUESTED** [review]

File: .github/workflows/cd.yml [hunk 5]
```diff
@@ -62,7 +62,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.7", "3.8", "3.9"]
+        python-version: ["3.8", "3.9", "3.10"]
```
[the remark this answers is not in the document]
Comment (Animatory) [reply .github/workflows/cd.yml:65]: It is time to move forward, and PyTorch 2.0 does not support 3.7

File: torchok/constructor/constructor.py (hunk 1 above)
Comment (Animatory) [reply torchok/constructor/constructor.py:332]: Because before, we were limited to adding new container augmentations. The common thing for all container transforms is the 'transforms' parameter requirement. So, now we can add custom container augmentations

File: torchok/models/heads/detection/__init__.py (hunk 2 above)
Comment (Animatory) [reply torchok/models/heads/detection/__init__.py:4]: Yes, but I am fixing this

File: torchok/constructor/config_structure.py (hunk 4 above)
Comment (Animatory) [reply torchok/constructor/config_structure.py:139]: No, they are compatible

File: torchok/constructor/constructor.py (hunk 1 above)
Comment (VladVin) [reply torchok/constructor/constructor.py:332]: Please add a comment about it here

File: torchok/models/heads/detection/__init__.py (hunk 2 above)
Comment (VladVin) [reply torchok/models/heads/detection/__init__.py:4]: What does it require to make it work? I'm estimating whether to put it into this PR

File: torchok/tasks/base.py (hunk 3 above)
Comment (VladVin) [reply torchok/tasks/base.py:129]: But we didn't track loss on each single step. It should be tracked with the configured frequency. So, how do configure that?

Comment (Animatory) [reply torchok/tasks/base.py:129]: Your CMD logs included step loss. The step size is determined in the config in TQDMProgressBar callback. Loss was tracked automatically in the previous version, but not it is not.

File: torchok/constructor/constructor.py (hunk 1 above)
Comment (Animatory) [reply torchok/constructor/constructor.py:333]: Added

File: torchok/tasks/base.py (hunk 3 above)
Comment (Animatory) [reply torchok/tasks/base.py:129]: Because I want loss to be displayed at every step

**Review by VladVin — APPROVED** [review]
